pkg/local_object_storage/metabase/exists.go

```diff
@@ -52,7 +57,7 @@ func (db *DB) Exists(prm ExistsPrm) (res ExistsRes, err error) {
 
 func (db *DB) exists(tx *bbolt.Tx, addr oid.Address) (exists bool, err error) {
 	// check graveyard first
-	switch inGraveyard(tx, addr) {
+	switch objectStatus(tx, addr, db.epochState.CurrentEpoch()) {
```

Comment:
Calling this inside the transaction can introduce unexpected logical errors: `CurrentEpoch` can return different values inside a single transaction. Can we provide this as a parameter instead?

Reply:
sure, fixed (although it required many changes in private methods)